<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [13]:
# Membuat dataframes dari dictionary

import pandas as pd

x = pd.DataFrame({'Name':['Fabio','Luiz'],
         'Club':['Roma','Shahktar'],
         'Age':['23','19'],
         'Pos':['ST','DF'],
         'Weekly_Wage':['54000','23000']})
print(x)

    Name      Club Age Pos Weekly_Wage
0  Fabio      Roma  23  ST       54000
1   Luiz  Shahktar  19  DF       23000


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [7]:
# Membaca data dan memasukannya kedalam suatu variabel

df_diamond_csv = pd.read_csv('diamonds-1.csv')

print(df_diamond_csv)

       Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
Series berbentuk daftar yang berbentuk vertikal tabular, sementara list dan dictionary adalah daftar horizontal 
- Apa perbedaan dataframe dan series
DataFrame dapat memuat 2 objek didalamnya, sementara series hanya satu
- Buatlah suatu data series

Jawab: 

In [14]:
# code here

y = pd.Series(['2','1','3','4','6'])

print(y)

0    2
1    1
2    3
3    4
4    6
dtype: object


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [8]:
# price-head
import pandas as pd
df_diamond_csv = pd.read_csv('diamonds-1.csv')

df_diamond_csv['price'].head(3)

0    326
1    326
2    327
Name: price, dtype: int64

In [7]:
#tail-color
df_diamond_csv['color'].tail(3)

53937    D
53938    H
53939    D
Name: color, dtype: object

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: Suatu metode dalam ilmu statistik yang berupaya untuk menampilkan gambaran umum perhitungan statistika pada sebuah data

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [11]:
# statistik deskriptif
import pandas as pd
df_diamond_csv = pd.read_csv('diamonds-1.csv')

df_diamond_csv['price'].describe()
df_diamond_csv['depth'].describe()
df_diamond_csv['carat'].describe()


count    53940.000000
mean         0.797940
std          0.474011
min          0.200000
25%          0.400000
50%          0.700000
75%          1.040000
max          5.010000
Name: carat, dtype: float64

Jawab: 
Price Mean: 3932.799722
Depth Std: 1.432621
Min, Max Carat: 0.200000, 5.010000